### PR TITLE
Do not shadow variable, use deprecated MAP_ANON if MAP_ANONYMOUS is not defined

### PR DESCRIPTION
**Pull request by paveljanik**
Two minor fixes after merging #8753.

### CODE REVIEW

File: src/support/lockedpool.cpp [hunk 1]
```diff
@@ -224,6 +224,13 @@ PosixLockedPageAllocator::PosixLockedPageAllocator()
     page_size = sysconf(_SC_PAGESIZE);
 #endif
 }
+
+// Some systems (at least OS X) do not define MAP_ANONYMOUS yet and define
+// MAP_ANON which is deprecated
```
Comment (laanwj) [inline src/support/lockedpool.cpp]: If MAP_ANON is deprecated, what is the new name?

Also: this passed Travis, how is that possible if it didn't exist on MacOSX?

**Comment by jonasschnelli** [comment]
utACK

FreeBSD manpage
```
 MAP_ANON           Map anonymous memory not associated with any specific
                    file.  The file descriptor used for creating MAP_ANON
                    must be -1.  The offset argument must be 0.

 MAP_ANONYMOUS      This flag is identical to MAP_ANON and is provided for
                    compatibility.
```

Linux manpage
```
MAP_ANONYMOUS
    The mapping is not backed by any file; the fd and offset arguments are ignored. This flag in conjunction with MAP_SHARED is implemented since Linux 2.4. 
MAP_ANON
    Alias for MAP_ANONYMOUS. Deprecated. 
```

OSX 10.11 manpage
```
     MAP_ANONYMOUS     Synonym for MAP_ANON.

     MAP_ANON          Map anonymous memory not associated with any specific file.  The offset argument is ignored.  Mac OS X spe-
                       cific: the file descriptor used for creating MAP_ANON regions can be used to pass some Mach VM flags, and can
                       be specified as -1 if no such flags are associated with the region.  Mach VM flags are defined in
                       <mach/vm_statistics.h> and the ones that currently apply to mmap are:
```

**Comment by laanwj** [comment]
So apparently all three OSes support MAP_ANONYMOUS? What am I missing?

**Comment by paveljanik** [comment]
My build machine is 10.10.

**Comment by jonasschnelli** [comment]
@paveljanik what does you 10.10s `man mmap` tells you about `MAP_ANONYMOUS`?
IMO travis compiles against 10.9... hmm..

**Comment by paveljanik** [comment]
@jonasschnelli Nothing... `man mmap | grep ANON`:

```
     MAP_ANON          Map anonymous memory not associated with any specific
                       cific: the file descriptor used for creating MAP_ANON
     [ENODEV]           MAP_ANON has not been specified and the file fd refers
                        the process.  MAP_ANON was specified and insufficient
```

**Comment by laanwj** [comment]
That's really strange. I really don't understand MacOSX' deprecation policy, from release to release it seems to oscillate wildly. In any case utACK.

**Comment by paveljanik** [comment]
Interesting:
```
$ grep ANON /usr/include/sys/mman.h
#define	MAP_ANON	0x1000	/* allocated from memory, swap space */
$ 
```

but

```
/Applications/Xcode.app//Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/sys/mman.h:#define	MAP_ANONYMOUS	MAP_ANON
/Applications/Xcode.app//Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/share/man/man2/mmap.2:.It Dv MAP_ANONYMOUS
/Applications/Xcode.app//Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS.sdk/usr/include/sys/mman.h:#define	MAP_ANONYMOUS	MAP_ANON
/Applications/Xcode.app//Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS2.1.sdk/usr/include/sys/mman.h:#define	MAP_ANONYMOUS	MAP_ANON
```

**Comment by paveljanik** [comment]
We are not alone: https://gmtk-trac.bitnamiapp.com/trac/gmtk/ticket/371 and others...

**Pull request merged**